addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/SpiderJob.java

```diff
@@ -92,6 +105,26 @@ public boolean applyCustomParameter(String name, String value) {
         return false;
     }
 
+    private void verifyIntValue(String name, String value, AutomationProgress progress) {
+        try {
+            Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            progress.error(
+                    Constant.messages.getString(
+                            "automation.error.options.badint", this.getType(), name, value));
```

Comment:
We should use `this.getName()` instead of `this.getType()` - that will give us the job name - in theory there could be 2 jobs with the same type and then it wouldnt be clear which one it applies to. there may well be other instances of this, and they might well be like that in the existing code ;)